helm/dendrite/templates/secrets.yaml

```diff
@@ -0,0 +1,116 @@
+{{ if (gt (len (.Files.Glob "appservices/*")) 0) }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Release.Name }}-appservices-conf
+  namespace: {{ .Release.Namespace }}
+type: Opaque
+data:
+{{ (.Files.Glob "appservices/*").AsSecrets | indent 2 }}
+{{ end }}
+{{ if and .Values.configuration.signing_key.create (not .Values.configuration.signing_key.existingSecret) }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  annotations:
+    helm.sh/resource-policy: keep
+  name: {{ .Release.Name }}-signing-key
+  namespace: {{ .Release.Namespace }}
+type: Opaque
+{{ end }}
+
+{{- $connectionString := print "postgresql://" .Values.configuration.database.user ":" .Values.configuration.database.password "@" .Values.configuration.database.host }}
+{{- if .Values.postgresql.enabled }}
+{{- $connectionString = print "postgresql://" .Values.postgresql.auth.username ":" .Values.postgresql.auth.password "@" .Release.Name "-postgresql/dendrite" -}}
+{{ end }}
+---
+apiVersion: v1
+kind: Secret
+type: Opaque
+metadata:
+  name: {{ .Release.Name }}-conf
+  namespace: {{ .Release.Namespace }}
+stringData:
+  dendrite.yaml: |
```

Comment:
It's kind of annoying to put the entire config YAML as a secret, as opposed to the actually secret stuff. It makes casually browsing deployments much more difficult.